File: core/src/main/java/uk/co/amlcurran/social/TimeCalculator.java
package uk.co.amlcurran.social;

import javax.annotation.Nonnull;

public interface TimeCalculator {

    @Nonnull
    Timestamp plusDays(int days, Timestamp time);

    int getDays(Timestamp time);

    @Nonnull
    Timestamp plusHours(Timestamp time, int hours);
}

File: android/core/src/main/java/uk/co/amlcurran/social/TimeCalculator.kt
package uk.co.amlcurran.social

interface TimeCalculator {

    fun plusDays(days: Int, time: Timestamp): Timestamp

    fun getDays(time: Timestamp): Int

    fun plusHours(time: Timestamp, hours: Int): Timestamp

    fun startOfToday(): Timestamp

}
